docs/USAGE-GUIDE.md

```diff
@@ -654,6 +654,18 @@ If the read end of the pipe is closed unexpectedly, writing to the pipe will rai
 a SIGPIPE signal. **s2n-tls does NOT handle SIGPIPE.** A SIGPIPE signal will cause
 the process to terminate unless it is handled or ignored by the application.
 
+**WARNING (TLS v1.2 ONLY):** Enabling Nagle's algorithm can increase handshake latency
+* Enabling Nagle's algorithm can increase the handshake latency. To avoid this behaviour, please disable Nagle's at the socket level; this can be done with the snippet displayed below. Please note this is only applicable to TLS v1.2.
+* If enabling Nagle's is required, then an increase in handshake latency is expected. This behavior occurs because `s2nc` does not coalesce multiple handshake messages into a record. In effect, after the `s2nc` sends the client key exchange record to the server, it waits until it receives a server ack before continuing the handshake.
+```
+#include <netinet/tcp.h>
+
+/* Example snippet to disable Nagle's for sockfd */
+int flag = 1;
+int result = setsockopt(sockfd, IPPROTO_TCP, TCP_NODELAY, (char*) &flag, sizeof(int));
+```
```

Comment:
Do we really need a code snippet for this? Couldn't you just say to use setsockopt with TCP_NODELAY? It's a fairly common use case.

Reply:
Ok. Will remove.